### PR TITLE
Make Django assertions available outside of a TestCase class

### PR DESCRIPTION
**Pull request by wgordon17**

### CODE REVIEW

**Comment by wgordon17** [comment]
This is my attempt at addressing some of the concerns in #232.

File: pytest_django/asserts.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return assertion_func
+
+
+__all__ = []
+asserts = set()
+asserts.update(
+    set(attr for attr in TestCase.__dict__ if attr.startswith('assert')),
+    set(attr for attr in SimpleTestCase.__dict__ if attr.startswith('assert')),
+    set(attr for attr in LiveServerTestCase.__dict__ if attr.startswith('assert')),
+    set(attr for attr in TransactionTestCase.__dict__ if attr.startswith('assert')),
```
Comment (graingert) [inline pytest_django/asserts.py]: use `vars(v)` over `v.__dict__`

File: docs/helpers.rst [hunk 2]
```diff
@@ -3,6 +3,18 @@
 Django helpers
 ==============
 
+Assertions
+----------
+
+All of Django's ``TestCase`` _assertions are available by adding the following
```
Comment (blueyed) [inline docs/helpers.rst]: > ``TestCase`` _assertions

Shouldn't that be `assertions_`?

File: pytest_django/asserts.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+"""
+Dynamically load all Django assertion cases and expose them for importing
```
Comment (blueyed) [inline pytest_django/asserts.py]:
```suggestion
Dynamically load all Django assertion cases and expose them for importing.
```

File: pytest_django/asserts.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+"""
+from django.test import (
+    TestCase, SimpleTestCase,
+    LiveServerTestCase, TransactionTestCase
+)
+
+
+def _wrapper(name):
+    def assertion_func(*args, **kwargs):
+        getattr(TestCase('run'), name)(*args, **kwargs)
```
Comment (blueyed) [inline pytest_django/asserts.py]: `TestCase('run')` should be cached I guess?!

File: tests/test_asserts.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+"""
+Tests the dynamic loading of all Django assertion cases
```
Comment (blueyed) [inline tests/test_asserts.py]:
```suggestion
Tests the dynamic loading of all Django assertion cases.
```

File: tests/test_asserts.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+import pytest
+import pytest_django
+
+from pytest_django.asserts import __all__ as asserts_all
+
+
+def _get_actual_assertions_names():
+    """
+    Returns list with names of all assertion helpers in Django
```
Comment (blueyed) [inline tests/test_asserts.py]:
```suggestion
    Returns list with names of all assertion helpers in Django.
```

File: pytest_django/asserts.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+asserts = set()
+asserts.update(
+    set(attr for attr in vars(TestCase) if attr.startswith('assert')),
+    set(attr for attr in vars(SimpleTestCase) if attr.startswith('assert')),
+    set(attr for attr in vars(LiveServerTestCase) if attr.startswith('assert')),
+    set(attr for attr in vars(TransactionTestCase) if attr.startswith('assert')),
+)
+
+for assert_func in asserts:
+    locals()[assert_func] = _wrapper(assert_func)
```
Comment (pakal) [inline pytest_django/asserts.py]: locals() is not meant to be always settable, maybe juste use globals() to make it clearer for newcomers?

**Comment by graingert** [comment]
Or replace sys.modules[__name__]

File: pytest_django/asserts.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+def _wrapper(name):
+    def assertion_func(*args, **kwargs):
+        getattr(test_case, name)(*args, **kwargs)
+
+    return assertion_func
+
+
+__all__ = []
+asserts = set()
```
Comment (pakal) [inline pytest_django/asserts.py]: "assertions_names" for more clarity?

**Comment by pakal** [comment]
Approach looks straightforward and handy, thanks for this PR  :)

**Comment by wgordon17** [comment]
@blueyed Any thoughts on when you'd want to merge this?

**Comment by blueyed** [comment]
Thanks for the update, I've pushed two improvements (better docs (star imports are discouraged), and changed it to use `functools.wraps` (for doc etc)).

**Comment by Mogost** [comment]
It looks so simple but so necessary. I look forward to the merge of this PR.

**Comment by wgordon17** [comment]
@blueyed Thanks so much for those improvements! 🙏

**Comment by blueyed** [comment]
Thanks! :)

**Comment by blueyed** [comment]
Released in 3.8.0.. :tada:

**Comment by premchalmeti** [comment]
@wgordon17  Any reason for not including `unittest.TestCase` asserts?

**Comment by wgordon17** [comment]
@premchalmeti, since this repo focuses on Django, it made more sense to focus only on the Django asserts. Additionally, most (if not all) the `unittest.TestCase` asserts are something that should be more appropriately handled by _traditional_ `assert` statements, as is the pytest philosophy.

**Pull request merged**